### PR TITLE
Add datastore-indexes.xml for easier deployment #7513

### LINKED ISSUE #7513: Add datastore-indexes.xml for easier deployment
Currently, we do not have a [datastore-indexes.xml](https://cloud.google.com/appengine/docs/standard/java/config/indexconfig) file in `src/main/webapp/WEB-INF/`, but instead rely on the autogenerated `src/main/webapp/WEB-INF/appengine-generated/datastore-indexes-auto.xml` which only gets updated index by index whenever the dev server performs an action that queries the datastore.

For the file to be fully populated, the entire test suite must first be run locally. This is required before deploying to a staging server, as the live Datastore requires indexes to be generated first before queries can succeed. If a deployment is attempted without this step, the resultant staging server is not fully functional - it will throw an exception whenever the user invokes an action that runs a query requiring an index that has not yet been generated.

By explicitly specifying all required indexes in `datastore-indexes.xml`, the testing step can be skipped and the project can be deployed from a fresh setup much more quickly.

### PR DESCRIPTION
**Pull request by whipermr5**
Fixes #7513

### CODE REVIEW

**Comment by whipermr5** [comment]
@wkurniawan07 It's an American/British English issue 😆 The ind(ex|ic)es correspond to my staging server, which passes the full test suite and has all the cron jobs running normally.

File: docs/development.md [hunk 1]
```diff
@@ -315,5 +315,6 @@ There are several files used to configure various aspects of the system.
 * `web.xml`: Contains the web server configuration, e.g servlets to run, mapping from URLs to servlets/JSPs, security constraints, etc.
 * `cron.xml`: Contains the cron jobs specification.
 * `queue.xml`: Contains the task queues configuration.
+* `datastore-indexes.xml`: Contains the Datastore indexes configuration.
```
Comment (LiHaoTan) [inline docs/development.md]: Should we add a `datastore-indexes-auto.xml` description here as well? See below.

File: src/main/webapp/WEB-INF/datastore-indexes.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<datastore-indexes autoGenerate="false">
```
Comment (LiHaoTan) [inline src/main/webapp/WEB-INF/datastore-indexes.xml]: Should we keep it as `autoGenerate="true"` so in the future if new indexes are needed in the future they can be automatically generated as well?

Comment (whipermr5) [inline src/main/webapp/WEB-INF/datastore-indexes.xml]: The thing is that they will be added to `appengine-generated/datastore-indexes-auto.xml`, which isn't source controlled. Setting to false will cause the dev server to fail, exactly like the live server. If someone is adding a new feature that requires a new index, they will have to update `datastore-indexes.xml`, which is source controlled. They can still do so by temporarily setting to true then copying the relevant ​lines from `appengine-generated/datastore-indexes-auto.xml` over.

Comment (LiHaoTan) [inline src/main/webapp/WEB-INF/datastore-indexes.xml]: Yea I just thought of that it's better to make it fail. Hence editing my comment to have a comment on it. Didn't know you can saw it LOL.

**Comment by whipermr5** [comment]
@LiHaoTan Just added the comment; have a look.

File: src/main/webapp/WEB-INF/datastore-indexes.xml [hunk 3]
```diff
@@ -1,4 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
+  - When adding a new index definition to support a new query, follow these steps:
+  - 1. Set autoGenerate to true.
+  - 2. Restart the dev server and run the new query.
+  -    The required index definition will be generated and saved in the file
+  -    "appengine-generated/datastore-indexes-auto.xml".
+  - 2. Copy the index definition from that file into this one.
```
Comment (LiHaoTan) [inline src/main/webapp/WEB-INF/datastore-indexes.xml]: Maybe can add the following so it is more complete?
`2. Copy the index definition from that file into this one (and tweak it if required).`

There is a typo in the numbering, should continue from `3.`

Comment (whipermr5) [inline src/main/webapp/WEB-INF/datastore-indexes.xml]: Oops! Thanks!

Comment (whipermr5) [inline src/main/webapp/WEB-INF/datastore-indexes.xml]: Based on our discussion, no need to add `(and tweak it if required)` as tweaking is only required under very special circumstances (complex queries that result in exploding indexes). Adding it in might confuse developers who are only developing simple queries, in which case the auto generated index definition is good enough.

**Comment by whipermr5** [comment]
@wkurniawan07 Sorry for the double request due to the bug!

**Pull request merged**